### PR TITLE
Port pmDAO, pmInfoDAO and AuthenticatedNSpecDAO to EasyDAO

### PR DESCRIPTION
**Pull request by KolombetM**
Port pmDAO, pmInfoDAO and AuthenticatedNSpecDAO to EasyDAO
https://nanopay.atlassian.net/browse/NP-229

### CODE REVIEW

File: src/services [hunk 1]
```diff
@@ -162,7 +164,11 @@ p({
   "name":"pmInfoDAO",
   "serve":true,
   "serviceScript":`
-    return new foam.nanos.pm.PMInfoMDAO().orderBy(foam.mlang.MLang.DESC(foam.nanos.pm.PMInfo.TOTAL_TIME));
+    dao = new foam.nanos.pm.PMInfoMDAO().orderBy(foam.mlang.MLang.DESC(foam.nanos.pm.PMInfo.TOTAL_TIME));
+    return new foam.dao.EasyDAO.Builder(x)
+      .setOf(foam.nanos.pm.PM.getOwnClassInfo())
+      .setDelegate(dao)
```
Comment (jlhughes) [inline src/services:170]:
```suggestion
      .setInnerDAO(dao)
```

File: src/services [hunk 2]
```diff
@@ -870,12 +876,14 @@ p({
...
-      .setDelegate(new foam.dao.NullDAO.Builder(x)
-      .setOf(foam.nanos.pm.PM.getOwnClassInfo())
-      .build())
+    dao = new foam.nanos.analytics.FoldManagerDAODecorator.Builder(x)
+      .setDelegate(new foam.dao.NullDAO(x, foam.nanos.pm.PM.getOwnClassInfo()))
       .setFm((foam.nanos.analytics.DAOFoldManager) x.get("pmFoldManager"))
       .build();
+    return new foam.dao.EasyDAO.Builder(x)
+      .setOf(foam.nanos.pm.PM.getOwnClassInfo())
+      .setDelegate(dao)
```
Comment (jlhughes) [inline src/services]:
```suggestion
      .setInnerDAO(dao)
```

File: src/services (hunk 1 above)
Comment (KolombetM) [reply src/services:170]: With this changes I'm getting "stack overflow" exception. As I remember, I need 
NullDAO to be the delegate for PMInfoMDAO but I can't setDelegate for PMInfoMDAO

Comment (jlhughes) [reply src/services:170]: ok, try setting 
.setNullify(true). <- need this else the inner dao will be another MDAO. 
.setDecorator(dao)

Comment (KolombetM) [reply src/services:170]: Still getting error

**Comment by KolombetM** [comment]
This daos will not be clustered

**Pull request closed without merging**